### ISSUE TITLE
DOC: doc-string validation broken

### ISSUE THREAD

**Issue by jreback** [pandas-dev/pandas#39688]
https://github.com/pandas-dev/pandas/pull/39685/checks?check_run_id=1857708224

```
==================================== ERRORS ====================================
__________ ERROR collecting scripts/tests/test_validate_docstrings.py __________
ImportError while importing test module '/home/runner/work/pandas/pandas/scripts/tests/test_validate_docstrings.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/share/miniconda/envs/pandas-dev/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
scripts/tests/test_validate_docstrings.py:5: in <module>
    import validate_docstrings
scripts/validate_docstrings.py:50: in <module>
    from numpydoc.validate import validate, Docstring  # isort:skip
E   ImportError: cannot import name 'Docstring' from 'numpydoc.validate' (/usr/share/miniconda/envs/pandas-dev/lib/python3.8/site-packages/numpydoc/validate.py)
=========================== short test summary info ============================
ERROR scripts/tests/test_validate_docstrings.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.54s ===============================
Error: Process completed with exit code 2.
````

cc @pandas-dev/pandas-core 
can we pin numpydoc?

**Comment by simonjayhawkins** [comment]
This is also occurring on 1.2.x, e.g. #39693


changing milestone

**Comment by simonjayhawkins** [comment]
moved to 1.2.5

**Comment by jorisvandenbossche** [comment]
I don't think this needs to be fixed for 1.2.x. It's not an error in our documentation, but a compatibility issue in our validation scripts. I think it's fine to target that for 1.3, and keep it pinned on 1.2.x

**Comment by simonjayhawkins** [comment]
sure. removing milestone.
